junit-platform-engine/src/main/java/org/junit/platform/engine/TestDescriptor.java

```diff
@@ -182,6 +183,24 @@ default void accept(Visitor visitor) {
 	}
 
 	/**
+	 * Does a logical level display name resolution for legacy reporting formats, instead of a source level name resolution.
+	 *
+	 * @return A LegacyReportingInfo containing the logical method and class display names.
+	 */
+	default LegacyReportingInfo getLegacyReportingInfo() {
+		final String displayName = getDisplayName();
+		final Optional<String> className = getParent().map(TestDescriptor::getDisplayName);
+
+		return new DefaultLegacyReportingInfo(displayName, className.orElse(null));
```

Comment:
This makes the degraph task fail in the ci, because org.junit.platform.engine now also references org.junit.platform.engine.support.descriptor, creating a cyclic dependency

I could just return an anomynous implementation here, but what is the preferred solution?